java/org/apache/catalina/filters/RateLimitFilter.java

```diff
@@ -63,6 +67,10 @@
  * information that it has, e.g. allow more requests to certain users based on roles, etc.
  * </p>
  * <p>
+ * The <code>exposeHeaders</code> allows output runtime information of rate limiter via response header, is disabled by
+ * default, only for http-api or debugging purpose. Those runtime information should not be accessible to the attackers.
+ * </p>
+ * <p>
```

Comment:
There are really two patches here. One to add the ExactRateLimiter and one to add support for "RateLimit header fields for HTTP". I think it would be better to do this as 2 PRs. There is a bit too much going on in this PR.

Reply:
I agree with @markt-asf.  It is much easier to review smaller PRs so breaking a large PR to smaller PRs, and even avoiding making unnecessary changes, make it much easier to review and accept.

For example, if this PR is for the ExactRateLimiter which does not use `java/org/apache/catalina/util/TimeBucketCounter.java` then changes to that class can be removed from here.